db/migrate/20240501135224_add_user_version_to_work_versions.rb

```diff
@@ -0,0 +1,7 @@
+class AddUserVersionToWorkVersions < ActiveRecord::Migration[7.0]
+  def change
+    add_column :work_versions, :user_version, :integer
+    add_index :work_versions, [:work_id, :user_version], unique: true
+    Work.all.each { |work| work.head.update(user_version: 1) }
```

Comment:
Does this make the migration irreversible? Maybe it's not worth caring about?

Reply:
Yes, we'd have to first comment this line out and then do `db:rollback`